websites/A/AnimeLib/presence.ts

```diff
@@ -0,0 +1,348 @@
+			"private-watching"
+		),
+		path = document.location.pathname;
+
+	let animeData: AnimeData,
+		userData: UserData,
+		characterData: CharacterData,
+		peopleData: PersonData,
+		collectionData: CollectionData,
+		reviewData: ReviewData,
+		teamData: TeamData,
+		publisherData: PublisherData;
+
+	switch (`${path}/`.split("/")[2]) {
+		case "":
+			presenceData.details = "Главная страница";
+			presenceData.state = "Так внимательно изучает...";
+			break;
+		case "anime":
+			animeData = await AnimeLib.getAnime(
+				path,
+				path.split("/")[3].split("-")[0]
+			).then(response => <AnimeData>response.data);
+
+			// Show anime watching in private mode if it's enabled, or enforce it when anime is RX rated (e.g hentai)
+			if (isWatchingPrivately || animeData.ageRestriction.id === 5) {
+				presenceData.details = "Приватный режим";
+				presenceData.state = "Явно что-то скрывает...";
+
+				break;
+			}
+
+			if (animeData.toast) {
+				const cover =
+						document.querySelector<HTMLImageElement>(".cover__img")?.src,
+					title = document.querySelector("h1")?.textContent,
+					altTitle = document.querySelector("h2")?.textContent;
+
+				if (cover && title && altTitle) {
+					presenceData.details = "Страница лицензированного аниме";
+					presenceData.state = `${title} (${altTitle})`;
+					presenceData.largeImageKey = cover;
+				}
+
+				break;
+			}
+
+			if (path.endsWith("/watch")) {
+				const video = document.querySelector("video"),
+					dub = document
+						.querySelector(".menu-item.is-active")
+						?.querySelector(".menu-item__text").textContent;
+
+				if (dub) {
+					presenceData.details = animeData.rus_name;
+					presenceData.state = `${
+						document.querySelector("[id^='episode'][class*=' ']")
+							?.textContent ?? "Фильм"
+					} | ${dub}`;
+					presenceData.largeImageKey = animeData.cover.default;
+
+					presenceData.smallImageKey = Assets.Pause;
+					presenceData.smallImageText = "На паузе";
+				}
+
+				if (video || iFrameVideo) {
+					if (video) {
+						[startTimestamp, endTimestamp] =
+							presence.getTimestampsfromMedia(video);
+						presenceData.smallImageKey = video.paused
+							? Assets.Pause
+							: Assets.Play;
+						presenceData.smallImageText = video.paused
+							? "На паузе"
+							: "Воспроизводится";
+
+						iFrameVideo = null;
+					} else {
+						[startTimestamp, endTimestamp] = presence.getTimestamps(
+							iFrameVideo.currentTime,
+							iFrameVideo.duration
+						);
+						presenceData.smallImageKey = iFrameVideo.paused
+							? Assets.Pause
+							: Assets.Play;
+						presenceData.smallImageText = iFrameVideo.paused
+							? "На паузе"
+							: "Воспроизводится";
+					}
+
+					presenceData.startTimestamp = startTimestamp;
+					presenceData.endTimestamp = endTimestamp;
+
+					if (video?.paused || iFrameVideo?.paused) {
+						delete presenceData.startTimestamp;
+						delete presenceData.endTimestamp;
+					}
+				}
+			} else {
```

Comment:
This should happen after:

Reply:
I'm not sure to what you refer.
```ts
				presenceData.details = "Страница аниме";
				presenceData.state = `${animeData.rus_name} (${
					animeData.eng_name ?? animeData.name
				})`;
				presenceData.largeImageKey = animeData.cover.default;
```
if you refer this, then no.
this part basically shows that you're on an anime's homepage: (example below)

<details>

![image](https://github.com/user-attachments/assets/1642d76c-907b-489e-802f-d4bdc84db794)


</details>

Reply:
Lgtm.